src/web/mjs/connectors/TenshiMoe.mjs

```diff
@@ -61,78 +62,100 @@ export default class TenshiMoe extends Connector {
     async _getMangasFromPage(page) {
         const uri = new URL('/anime?page=' + page, this.url);
         const request = new Request(uri, this.requestOptions);
-        const data = await this.fetchDOM(request, 'ul.loop.anime-loop li a');
-        return data.map(element => {
+        let data = await this.fetchDOM(request, 'div#app');
+        data = JSON.parse(data[0].dataset['page']);
+        return data.props.anime_list.data.map(element => {
             return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                id: element.slug,
                 title: element.title.trim()
             };
         });
     }
 
     async _getChapters(manga) {
         let chapterlist = [];
-        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
-        const data = await this.fetchDOM(request, 'ul.pagination li:nth-last-of-type(2) a');
-        const pageCount = data.length > 0 ? parseInt(data[0].text): 1;
+        const uri = new URL('/anime/'+manga.id, this.url);
+        let request = new Request( uri.href, this.requestOptions );
+        let token = await Engine.Request.fetchUI( request,
+            `new Promise( resolve =>
+         resolve(document.cookie) 
+          )` );
+
+        token = decodeURIComponent(token.match(/XSRF-TOKEN=([\S]+)/)[1]);
+
+        request = new Request(new URL('/anime/'+manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'div#app');
+        data = JSON.parse(data[0].dataset['page']);
+        const pageCount = data.props.episode_list.meta.last_page;
+        this.version = data.version;
```

Comment:
Seems like `version` could make a local variable instead of global and provided as parameter to `_getchaptersFromPage(...)` the same way as it is done with `token`